cosmetics-web/app/analyzers/read_data_analyzer.rb

```
@@ -9,7 +9,11 @@ def initialize(blob)
   end
 
   def self.accept?(given_blob)
-    return false unless given_blob.present? && given_blob.metadata["safe"]
+    return false if given_blob.blank?
+
+    if ENV.fetch('ANTIVIRUS_ENABLED', 'true') == 'true'
+      return false unless given_blob.metadata["safe"]
```

Are we sure we always have the `safe` key within `given_blob.metadata`.

---

That's what the existing code assumes - the key is added by the antivirus analyzer. This just removes that condition if you've set `ANTIVIRUS_ENABLED=false`.

---

I appreciate that, although if the AV changes this, which we seem to have little control over, we'll have to do some digging to work out what's wrong.

May if it's not present we could throw an exception with a helpful error message. Or if it's not vital, just return false 🤷‍♂

---

I'm a bit confused by your comment... 😕 

We need to continue to check that "safe" has been added to the metadata (which is done by our code [here](https://github.com/UKGovernmentBEIS/beis-opss/blob/master/cosmetics-web/app/analyzers/anti_virus_analyzer.rb#L13)), in order to make sure the files have been scanned for viruses before being parsed by our XML parser.

This change just no longer applies this check if you've set `ANTIVIRUS_ENABLED=false` so that we can run the application and the Sidekiq queue locally without having the antivirus running.

---

Ah, cool, if we have control over it, that's slightly more comforting.

We should still not assume that we're getting the metadata in this format though. We should make sure that it actually exists instead of expecting it to.

---

I still don't quite follow. Are you suggestion we should be checking the actual value in the Hash rather than just that there's a `safe` key?

---

Perhaps. I've not looked at what values are actually stored under that key. This PR doesn't change the existing condition logic, it just gives developers the opportunity to turn the condition off. We could review how the antivirus scanner does & should work as a separate task?